apps/client/src/components/globalSettings/PlotSelector.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+    <div>
+        <div class="q-item-section__right">
+            <q-btn
+                class="gt-xs"
+                size="12px"
+                flat
+                dense
+                round
+                icon="filter_alt"
+                color="grey-7"
+                @click="addFilter"
+            />
+        </div>
+        <UnivariateCellPlot
+            v-for="plotName in plotNames"
+            :key="plotName"
+            :plot-name="plotName"
+            :plot-brush="plotBrush"
+            @selection-change="handleSelectionChange"
+        ></UnivariateCellPlot>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { ref } from 'vue';
+import { QBtn } from 'quasar';
+import * as vg from '@uwdata/vgplot';
+import { storeToRefs } from 'pinia';
+import UnivariateCellPlot from './UnivariateCellPlot.vue';
+import { useFilterStore } from '@/stores/filterStore';
+import { useSelectionStore } from '@/stores/selectionStore';
+
+interface Selections {
+    [key: string]: [number, number] | undefined;
+}
+interface SelectionChangeEvent {
+    plotName: string;
+    range: [number, number] | null;
+}
+
+const plotNames = ref(['A', 'y', 'mass', 'time', 'x']);
+
+const plotBrush = vg.Selection.intersect();
+vg.Selection.crossfilter();
+
+const currentSelections = ref<Selections>({});
+const filterStore = useFilterStore();
+const selectionStore = useSelectionStore();
+
+// Invoked by selectionChange event in UnivariateCellPlot
+const handleSelectionChange = (event: SelectionChangeEvent) => {
+    const { plotName, range } = event;
+    if (range && range[0] !== undefined && range[1] !== undefined) {
+        currentSelections.value[plotName] = range;
+        selectionStore.updateSelection(plotName, [
+            range[0].toFixed(2),
```

Comment:
I would not convert to string here. Instead I would store the actual number in the selectionStore (will have to update types in that file). Then only format the string when you display it (I assume that's why you want two decimal places?).